components/PersonWidget/PersonWidget.spec.tsx

```diff
@@ -54,6 +54,40 @@ describe('PersonWidget', () => {
     ).toBeNull();
   });
 
+  it('should return the last address if there are multiple addresses saved on the person', () => {
+    const firstAddress = {
+      addressLines: '100 Example Street',
+      postCode: 'E1 8HW',
+    };
+
+    const newAddress = {
+      addressLines: '123 TestCode Lane',
+      postCode: 'SE1 7DT',
+    };
+
+    const resident = residentFactory.build();
+    resident.addresses = [firstAddress, newAddress];
+
+    const { getByTestId, asFragment } = render(
+      <PersonWidget person={resident} />
+    );
+
+    expect(getByTestId('resident-last-address')).toHaveTextContent(
+      '123 TestCode Lane'
+    );
+    expect(getByTestId('resident-last-address')).toHaveTextContent('SE1 7DT');
+
+    expect(getByTestId('resident-last-address')).not.toHaveTextContent(
+      '100 Example Street'
+    );
+
+    expect(getByTestId('resident-last-address')).not.toHaveTextContent(
+      'E1 8HW'
+    );
+
+    expect(asFragment()).toMatchSnapshot();
```

Comment:
Small nitpick – the `expect` statements in your test case above are pretty explicit, you can probably skip having the snapshot too

Reply:
I see. 🤔 Thanks!
I wasn't sure if needed the snapshot 😅 . When's the best time to use a snapshot?

Reply:
There's a few schools of thoughts around snapshots. My general rules of thumb are:

* Snapshots are good for high level pages, where you want to be alerted when a smaller change in a sub-component is likely to have an impact
* Snapshots are good when it's hard to be specific with your `expect` statements, perhaps it's a complex DOM element (like a `<select>`, `<form>` or something)
* Snapshots can be good for complex data objects (e.g. a nested set of JSON) as it's just doing a string match, so you don't have to check every property explicitly

* Snapshots are bad when there are clear and concise `expect` statements which more clearly define the intent of the code under test
* Snapshots are bad when the DOM is going to change _a lot_ (you'll just be constantly updating them, better to aim for some general expect statements)
* Snapshots are bad when there isn't a lot of DOM – they encourage lazy code reviews (not maliciously, but they're easy to skip over!), better to have explicit expect statements

As with most things, used in moderation, they can be effective!